consensus/src/dag/tests/reliable_broadcast_tests.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::dag::reliable_broadcast::{
+    BroadcastStatus, DAGMessage, DAGNetworkSender, ReliableBroadcast,
+};
+use crate::network_interface::ConsensusMsg;
+use anyhow::bail;
+use aptos_consensus_types::common::Author;
+use aptos_infallible::Mutex;
+use aptos_types::validator_verifier::random_validator_verifier;
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use std::collections::{hash_map::Entry, HashMap, HashSet};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::oneshot;
+
+#[derive(Serialize, Deserialize, Clone)]
+struct TestMessage(Vec<u8>);
+
+impl DAGMessage for TestMessage {
+    fn from_network_message(msg: ConsensusMsg) -> anyhow::Result<Self> {
+        match msg {
+            ConsensusMsg::DAGTestMessage(payload) => Ok(Self(payload)),
+            _ => bail!("wrong message"),
+        }
+    }
+
+    fn into_network_message(self) -> ConsensusMsg {
+        ConsensusMsg::DAGTestMessage(self.0)
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+struct TestAck;
+
+impl DAGMessage for TestAck {
+    fn from_network_message(_: ConsensusMsg) -> anyhow::Result<Self> {
+        Ok(TestAck)
+    }
+
+    fn into_network_message(self) -> ConsensusMsg {
+        ConsensusMsg::DAGTestMessage(vec![])
+    }
+}
+
+struct TestBroadcastStatus {
+    threshold: usize,
+    received: HashSet<Author>,
+}
+
+impl BroadcastStatus for TestBroadcastStatus {
+    type Message = TestMessage;
+    type Ack = TestAck;
+    type Aggregated = HashSet<Author>;
+
+    fn empty(receivers: Vec<Author>) -> Self {
+        Self {
+            threshold: receivers.len(),
+            received: HashSet::new(),
+        }
+    }
+
+    fn add(&mut self, peer: Author, _ack: Self::Ack) -> anyhow::Result<Option<Self::Aggregated>> {
```

Comment:
Do we expect to return an Err() result here? If not, this can be just an Option<>

Reply:
I think so, like wrong signature for example